### PR TITLE
Set operation type

### PR DESCRIPTION
**Pull request by ZeHiro**

### CODE REVIEW

**Comment by ZeHiro** [comment]
@bnjbvr  This one seems good.

File: client/components/OperationList.js [hunk 1]
```diff
@@ -71,6 +71,65 @@ class CategorySelectComponent extends React.Component {
...
+        var label = store.operationTypeToLabel(selectedId);
+
+        if (!this.state.editMode) {
+            return (<span onClick={this.switchToEditMode.bind(this)}>{label}</span>)
+        }
+
+        // On the first click in edit mode, categories are already loaded.
+        // Every time we reload categories, we can't be in edit mode, so we can
+        // just synchronously retrieve categories and not need to subscribe to
+        // them.
```
Comment (bnjbvr) [inline client/components/OperationList.js]: nit: update comment

File: client/backend.js [hunk 2]
```diff
@@ -212,5 +212,17 @@ module.exports = {
             success: cb,
             error: xhrError
         });
+    },
+    
+    setTypeForOperation: function(operationId, operationTypeId, cb) {
```
Comment (bnjbvr) [inline client/backend.js]: new backend functions should use the Promise model, but it's fine to let as is; I plan doing a full pass at some point anyways.

File: client/components/OperationList.js [hunk 3]
```diff
@@ -71,6 +71,65 @@ class CategorySelectComponent extends React.Component {
...
+        var options = store.getOperationTypes().map(function(c) {
+            if (c.id == NONE_OPERATION_TYPE_ID) {
```
Comment (bnjbvr) [inline client/components/OperationList.js]: nit: `if (c.id === NONE_OPERATION_TYPE_ID) {`

File: server/controllers/operations.coffee [hunk 4]
```diff
@@ -39,8 +39,8 @@ module.exports.update = (req, res) ->
     attr = req.body
 
     # For now, we can only update the category id of an operation.
```
Comment (bnjbvr) [inline server/controllers/operations.coffee]: nit: stale comment, please update

File: client/store.js [hunk 5]
```diff
@@ -782,6 +791,16 @@ export let Actions = {
...
         });
     },
+    
+    SetOperationType(operation, typeId) {
+        assert(operation instanceof Operation, 'SetOperationType expects an Operation as the first argument');
+        assert(typeof typeId === 'string', 'SetOperationType expects a String category id as the second argument');
+        flux.dispatch({
+            type: Events.user.updated_type_of_operation,
+            operation,
+            typeId: typeId
```
Comment (bnjbvr) [inline client/store.js]: nit: `typeId,`

File: client/store.js [hunk 6]
```diff
@@ -782,6 +791,16 @@ export let Actions = {
             accessId: account.bankAccess
         });
     },
+    
+    SetOperationType(operation, typeId) {
+        assert(operation instanceof Operation, 'SetOperationType expects an Operation as the first argument');
+        assert(typeof typeId === 'string', 'SetOperationType expects a String category id as the second argument');
```
Comment (bnjbvr) [inline client/store.js]: nit: s/category/operation type/ in the error mesage

File: client/components/OperationList.js (hunk 3 above)
Comment (bnjbvr) [inline client/components/OperationList.js]: Is the intent to block the user from setting "None" as the operation type? In that case, you might as well go with:

```
var options = store.getOperationTypes().filter((t) => t.id !== NONE_OPERATION_TYPE_ID).map(...)
```

and the map doesn't need to contain the special case with the `if`.

File: client/components/OperationList.js (hunk 1 above)
Comment (ZeHiro) [inline client/components/OperationList.js]: OK

**Comment by bnjbvr** [comment]
Looks good! A few small remarks that I'd like to see addressed before merging. Thanks for the PR!

File: client/backend.js (hunk 2 above)
Comment (ZeHiro) [inline client/backend.js]: I'll let it like that as I don't knoew the promise model. If you point to the API, I can correct now.

File: client/components/OperationList.js (hunk 3 above)
Comment (ZeHiro) [inline client/components/OperationList.js]: OK

File: server/controllers/operations.coffee (hunk 4 above)
Comment (ZeHiro) [inline server/controllers/operations.coffee]: OK

File: client/store.js (hunk 5 above)
Comment (ZeHiro) [inline client/store.js]: line 801? If you loog at line 811, the ',' is not here. Do I update all the others?

File: client/store.js (hunk 6 above)
Comment (ZeHiro) [inline client/store.js]: OK

File: client/store.js (hunk 5 above)
Comment (bnjbvr) [inline client/store.js]: You're right, it shouldn't need the trailing comma at the end.

**Comment by ZeHiro** [comment]
Except form the backend.js comment I can't resolve, all the comments are taken into account

**Comment by ZeHiro** [comment]
What about the style of the "listes déroulantes" (as for categories), they are not so sexy. Do we use the class of the "liste déroulante" used for the research field "categories"

**Comment by bnjbvr** [comment]
You're right; I'll change that (I've already started adding changes over your commit locally, will push soon).

**Pull request merged**